lib/ui.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'messages'
+
+class UserInterface
+    def initialize
+        @message = Message.new
+    end
+
+    def start_game
+        puts @message.lookup(:welcome)
+        puts @message.lookup(:instructions)
+        puts display_board
+    end
+
+    def get_spot
+        puts @message.lookup(:get_spot)
+        spot = gets.chomp.to_i
+        if valid_spot?(spot)
+            return spot
+        else
+            puts @message.lookup(:invalid_input)
+            get_spot
+        end
+    end
+
+    def valid_spot?(spot)
+        (1..9).to_a.include?(spot)
+    end
+end
```

Comment:
This is telling you to add an empty line at the end of your file :) I believe vim should do this for you unless you remove it manually. You can fix it by adding the extra line and confirming all your files have one extra line at the EoF (end of file).

Reply:
Thank you for pointing this out! This actually helped me realize that I may have turned this off and never turned it back on during my vim tutorials, so I'm going to look into that today.